src/Ilios/CoreBundle/DataFixtures/ORM/AbstractFixture.php

```diff
@@ -0,0 +1,127 @@
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * A Doctrine fixture based data loader base class that populates the DB from data files.
+ *
+ * Class AbstractFixture
+ * @package Ilios\CoreBundle\DataFixtures\ORM
+ */
+abstract class AbstractFixture extends DataFixture implements
+    FixtureInterface,
+    ContainerAwareInterface
+{
+    /**
+     * @var string
+     * Doubles as identifier for this fixture's data file and entity references.
+     */
+    protected $key;
+
+    /**
+     * @var boolean
+     * Set to TRUE if the loaded fixture should be held on for reference.
+     */
+    protected $storeReference;
+
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+
+
+    /**
+     * @param string $key
+     * @param boolean $storeReference
+     */
+    public function __construct($key, $storeReference = true)
+    {
+        $this->key = $key;
+        $this->storeReference = $storeReference;
+    }
+
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setContainer(ContainerInterface $container = null)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load(ObjectManager $manager)
+    {
+
+        $fileName = $this->getKey() . '.csv';
+        $path = $this->getDataFilePath($fileName);
+
+        $first = true;
+        if (($handle = fopen($path, 'r')) !== false) {
+            while (($data = fgetcsv($handle)) !== false) {
+                // step over the first row
+                // since it contains the field names
+                if ($first) {
+                    $first = false;
+                    continue;
+                }
+                $entity = $this->createEntity($data);
+                $manager->persist($entity);
+
+                // honor the given entity identifiers.
+                // @link http://www.ens.ro/2012/07/03/symfony2-doctrine-force-entity-id-on-persist/
+                $metadata = $manager->getClassMetaData(get_class($entity));
+                $metadata->setIdGeneratorType(ClassMetadata::GENERATOR_TYPE_NONE);
+
+                if ($this->storeReference) {
+                    $this->addReference($this->getKey() . $entity->getId(), $entity);
+                }
+                $manager->flush();
+            }
+            fclose($handle);
+        }
+    }
+
+    /**
+     * Finds and return the absolute path to a given data file.
+     * @param string $fileName name of the data file.
+     * @return string the absolute path.
+     */
+    protected function getDataFilePath($fileName)
+    {
+        /**
+         * @var FileLocator $fileLocator
+         */
+        $fileLocator = $this->container->get('file_locator');
+
+        // TODO: pull this hardwired path to the data files out into configuration. [ST 2015/08/26]
```

Comment:
Would the point of pulling this out be to allow an implementer to use their own fixtures?  Otherwise I think maybe it could be a constant in this class, but we're not likely to want to move it around.